packages/codegen-ui/lib/utils/form-component-metadata.ts

```diff
@@ -57,6 +65,20 @@ export const mapFormMetadata = (form: StudioForm, formDefinition: FormDefinition
       if ('dataType' in config && config.dataType) {
         metadata.dataType = config.dataType;
       }
+      // create mapping for field name collisions
+      let filteredFieldName = filterFieldName(name);
+      let count = 1;
+      if (mappedFieldVariableNames.has(filteredFieldName.toLowerCase()) && !name.includes('.')) {
+        filteredFieldName += count;
+        while (mappedFieldVariableNames.has(filteredFieldName.toLowerCase())) {
+          count += 1;
+          filteredFieldName = filteredFieldName.slice(0, filteredFieldName.length - 1) + count;
```

Comment:
this could break if count starts getting into double digits. Rather than repeatedly appending and slicing off the count as you continue to check for conflicting names, line72 could be `const prospectiveNewName = filteredFieldName + count;` and reassigned on this line. Then you only append to `filteredFieldName` after it's certain that it will be unique.

Reply:
Oh yeah good catch